Sources/Controllers/AlloyScannerViewController.swift

```diff
@@ -0,0 +1,378 @@
+
+  func createOverlay() -> UIView {
+    // Create the view and add the blur
+    let overlayView = UIView(frame: view.frame)
+    overlayView.backgroundColor = UIColor.black.withAlphaComponent(0.4)
+
+    // Frame
+    let width = view.frame.width * BoundingBoxConstants.horizontalPadding
+    let height = view.frame.height * BoundingBoxConstants.verticalPadding
+    let centerX = view.center.x - (width / BoundingBoxConstants.middle)
+    let centerY = view.center.y - (height / BoundingBoxConstants.middle)
+
+    // Add rectangle that will be our rect of interest
+    let path = CGMutablePath()
+    path.addRoundedRect(
+      in: CGRect(x: centerX,
+                 y: centerY,
+                 width: width,
+                 height: height),
+      cornerWidth: BoundingBoxConstants.cornerSize,
+      cornerHeight: BoundingBoxConstants.cornerSize)
+    path.closeSubpath()
+
+    // Add border to our rectangle
+    borderShapeLayer = CAShapeLayer()
+    guard let borderShape = self.borderShapeLayer else {
+      return UIView()
+    }
+    borderShape.path = path
+    borderShape.lineWidth = BoundingBoxConstants.borderWidth
+    borderShape.strokeColor = self.configuration.focusViewStrokeColor.cgColor
+    overlayView.layer.addSublayer(borderShape)
+
+    path.addRect(CGRect(origin: .zero, size: overlayView.frame.size))
+
+    let maskLayer = CAShapeLayer()
+    maskLayer.backgroundColor = UIColor.black.cgColor
+    maskLayer.path = path
+    maskLayer.fillRule = CAShapeLayerFillRule.evenOdd
+
+    overlayView.layer.mask = maskLayer
+    overlayView.clipsToBounds = true
+
+    return overlayView
+  }
+}
+
+// MARK: - Bounding box Animations
+extension AlloyScannerViewController {
+  typealias AnimationConstants = AlloyScannerConstants.ReadingAnimation
+
+  private func startReadingAnimation() {
+    borderShapeLayer?.removeAllAnimations()
+    let animation = CABasicAnimation(keyPath: #keyPath(CALayer.opacity))
+    animation.fromValue = AnimationConstants.fromValue
+    animation.toValue = AnimationConstants.toValue
+    animation.duration = AnimationConstants.duration
+    animation.repeatCount = .infinity
+    animation.autoreverses = true
+    animation.timingFunction = CAMediaTimingFunction(name: .easeInEaseOut)
+    borderShapeLayer?.add(animation, forKey: #keyPath(CALayer.opacity))
+
+    view.layoutIfNeeded()
+  }
+
+  private func stopReadingAnimation() {
+    borderShapeLayer?.removeAllAnimations()
+  }
+
+  private func stopAnimations() {
+    borderShapeLayer?.removeAllAnimations()
+  }
+}
+
+// MARK: - Layout
+extension AlloyScannerViewController {
+  typealias LayoutConstants = AlloyScannerConstants.LayoutConstants
+
+  func addFocusView() {
+    self.focusView = createOverlay()
+    if let focusView = self.focusView {
+      view.addSubview(focusView)
+      self.view.bringSubviewToFront(focusView)
+    }
+  }
+
+  private func addFlashButton() {
+    self.view.addSubview(flashButton)
+
+    NSLayoutConstraint.activate([
```

Comment:
why not use Plangrid extensions?

Reply:
Unfortunately as it is a framework, I don't have access to the PlanGrid code